File: spring-rest-configuration-autodiscover.scala/src/main/scala/demo/service/core/PropertyIntegerService.scala
package demo.service.core

abstract class PropertyIntegerService extends AbstractPropertyService[Int] {

  override def getType: Type.Value = Type.INTEGER

  override def fromString(input: String): Int =
    if (input == null) getDefaultValue else Integer.valueOf(input)

  override def toString(input: Int): String =
    if (input == null) getDefaultValue.toString else String.valueOf(input)
}

File: spring-rest-configuration-autodiscover.scala/src/main/kotlin/demo/service/core/PropertyIntegerService.kt
package demo.service.core

abstract class PropertyIntegerService : AbstractPropertyService<Int>() {

    override fun getType() = Type.INTEGER

    public override fun fromString(input: String?): Int =
            if (input == null) getDefaultValue() else Integer.valueOf(input)

    public override fun toString(input: Int?): String =
            input?.toString() ?: getDefaultValue().toString()
}
